test/integration/good/code-gen/cl.expected

```diff
@@ -1298,54 +1304,65 @@ class optimize_glm_model final : public model_base_crtp<optimize_glm_model> {
       stan::math::fill(alpha_v, std::numeric_limits<double>::quiet_NaN());
       
       current_statement__ = 1;
-      alpha_v = in__.vector(k);
+      assign(alpha_v,
+        in__.template read<Eigen::Matrix<local_scalar_t__, -1, 1>>(k),
+        "assigning variable alpha_v");
       Eigen::Matrix<double, -1, 1> beta;
       beta = Eigen::Matrix<double, -1, 1>(k);
       stan::math::fill(beta, std::numeric_limits<double>::quiet_NaN());
       
       current_statement__ = 2;
-      beta = in__.vector(k);
+      assign(beta,
+        in__.template read<Eigen::Matrix<local_scalar_t__, -1, 1>>(k),
+        "assigning variable beta");
       Eigen::Matrix<double, -1, 1> cuts;
       cuts = Eigen::Matrix<double, -1, 1>(k);
       stan::math::fill(cuts, std::numeric_limits<double>::quiet_NaN());
       
       current_statement__ = 3;
-      cuts = in__.vector(k);
+      assign(cuts,
+        in__.template read<Eigen::Matrix<local_scalar_t__, -1, 1>>(k),
+        "assigning variable cuts");
       double sigma;
       sigma = std::numeric_limits<double>::quiet_NaN();
       
       current_statement__ = 4;
-      sigma = in__.scalar();
-      current_statement__ = 4;
-      sigma = stan::math::lb_constrain(sigma, 0);
+      sigma = in__.template read_constrain_lb<local_scalar_t__, jacobian__>(
+                0, lp__);
       double alpha;
       alpha = std::numeric_limits<double>::quiet_NaN();
       
       current_statement__ = 5;
-      alpha = in__.scalar();
+      alpha = in__.template read<local_scalar_t__>();
       double phi;
       phi = std::numeric_limits<double>::quiet_NaN();
       
       current_statement__ = 6;
-      phi = in__.scalar();
+      phi = in__.template read<local_scalar_t__>();
       Eigen::Matrix<double, -1, -1> X_p;
       X_p = Eigen::Matrix<double, -1, -1>(n, k);
       stan::math::fill(X_p, std::numeric_limits<double>::quiet_NaN());
       
       current_statement__ = 7;
-      X_p = in__.matrix(n, k);
+      assign(X_p,
+        in__.template read<Eigen::Matrix<local_scalar_t__, -1, -1>>(n, k),
+        "assigning variable X_p");
```

Comment:
Do these have to go through assigns?

Reply:
I'd like to avoid it but idk how. Since we are guaranteed that the deserializer has enough values in it for each parameter this could be 

```
      Eigen::Matrix<local_scalar_t__,, -1, -1> X_p = in__.template read<Eigen::Matrix<local_scalar_t__, -1, -1>>(n, k);
```

The assignment is generated [here](https://github.com/stan-dev/stanc3/pull/856/files#diff-a1f5ff85d2b828c19a91e01d5c1e8caaf75c47e8af9bb205a3e0ed486a7e064fR253). Would we need to change that `Fixed.pattern` to a `Decl`? I can't figure out how to do deceleration and assignment on the same line. But doing that would save both memory and computational time

Reply:
I pushed a change that removes that, feel free to remove it if you dont like it.